lib/model.js

```diff
@@ -948,12 +953,13 @@ Connection.from = (url, callback) => {
       return callback(error);
     }
 
+    const extraParsed = URL.parse(unescapedUrl, true);
     const attrs = Object.assign(
       {},
       {
         hosts: parsed.hosts,
-        hostname: parsed.hosts[0].host,
-        port: parsed.hosts[0].port,
+        hostname: isSrvRecord ? extraParsed.hostname : parsed.hosts[0].host,
```

Comment:
Hmm I wonder if this is more legible getting rid of the inline ternaries and move up to a single `if(isSrvRecord){...}`. @alenakhineika what do you think?

Reply:
Updated! I agree, looks better with `if`, cleaner.